### PR TITLE
Org SSO Page updates

### LINKED ISSUE #15961: [SSO] Support edit and remove of OIDC clients
The preliminary UI provided with https://github.com/gitpod-io/gitpod/issues/14278 lacks the implementation for edit/removal of OIDC clients.

Once the API is in place, it should be feasible to complement the features.

Note: the result still remains "preliminary", it's just to enable SSO.

### PR DESCRIPTION
**Pull request by selfcontained**
## Description

Makes various updates to the Org Settings => SSO page.

* Adds Delete support for OIDC clients
* Adds Edit support to the UI for OIDC clients (api needs implementing still)
* Updates page to use react-query
* Updates page to take advantage of existing components similar to Git Auth.

## Related Issue(s)

Relates to #15961

## How to test

**Preview Env:** https://bmh-org-se0fee9b85fb.preview.gitpod-dev.com/workspaces
* Create an Org, visit the SSO page
* Create a client.
* Try to edit it. UI should populate w/ client data, but saving should product an error as the API isn't implemented yet.
* Try to delete the client

## Release Notes

```release-note
NONE
```

## Build Options:

- [ ] /werft with-werft
      Run the build with werft instead of GHA
- [ ] leeway-no-cache
- [ ] /werft no-test
      Run Leeway with `--dont-test`

<details>
<summary>Publish Options</summary>

- [ ] /werft publish-to-npm
- [ ] /werft publish-to-jb-marketplace
</details>

<details>
<summary>Installer Options</summary>

- [ ] with-ee-license
- [ ] with-dedicated-emulation
- [ ] with-ws-manager-mk2
- [ ] workspace-feature-flags
  Add desired feature flags to the end of the line above, space separated
</details>

#### Preview Environment Options:
- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [x] /werft with-preview
- [ ] /werft with-large-vm
- [ ] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [ ] /werft with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`

### CODE REVIEW

**Comment by roboquat** [comment]
@selfcontained: Adding the "do-not-merge/release-note-label-needed" label because no release-note block was detected, please follow our [release note process](https://git.k8s.io/community/contributors/guide/release-notes.md) to remove it.

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

File: components/dashboard/src/teams/git-integrations/GitIntegrations.tsx
Comment (selfcontained) [inline components/dashboard/src/teams/git-integrations/GitIntegrations.tsx:22]: Adding a heading/subheading to the content of the git auth page so it's a bit more consistent w/ the SSO page layout.

File: components/dashboard/src/components/Modal.tsx
Comment (selfcontained) [inline components/dashboard/src/components/Modal.tsx:153]: Positioning off of the bottom allows it to grow taller if needed (still some styling issues for long error messages, but this should be a small improvement).

File: components/dashboard/src/teams/GitIntegrationsPage.tsx
Comment (selfcontained) [inline components/dashboard/src/teams/GitIntegrationsPage.tsx:12]: Wrapper component no longer needed, as the loading state is handled in `<OrgSettingsPage/>` now.

File: components/public-api-server/pkg/apiv1/oidc.go
Comment (selfcontained) [inline components/public-api-server/pkg/apiv1/oidc.go:304]: Noticed `oidcConfig.issuer` wasn't returned in our responses for clients. This returns it, but would like to make sure it wasn't intentional that we weren't returning it.

Comment (AlexTugarev) [reply components/public-api-server/pkg/apiv1/oidc.go:304]: It was probably overlooked, because it wasn't used anywhere. Thanks for adding it 🙏🏻

File: components/dashboard/src/teams/sso/OIDCClientConfigModal.tsx
Comment (AlexTugarev) [inline components/dashboard/src/teams/sso/OIDCClientConfigModal.tsx:92]: It would be great to surface those errors.

Especially for the backend validation those errors are crucial to identify connectivity problems between Gitpod's backend and the IdP services. The client side validation is nice and a good improvement, but if the backend cannot connect to the services, the auth processes will break too late on receiving the callback from IdP.

For example if you enter some valid URL, which might point to non-existing host, we see the following error from backend. I agree it's a bit too long, we might need to tweak that (on backend) as well. While this is about non exiting host, same applies for hosts which are not reachable from the network the backend services are running in, so basically it's a whole class of connectivity issues.

> Get "https://google234444444.com": dial tcp: lookup google234444444.com on 10.43.0.10:53: no such host

File: components/dashboard/src/teams/GitIntegrationsPage.tsx
Comment (gtsiolis) [reply components/dashboard/src/teams/GitIntegrationsPage.tsx:12]: **issue**: I wanted to mention this in https://github.com/gitpod-io/gitpod/pull/16850, but the goal of the settings page was to keep the _Organizations Settings_ in the header around for all pages under settings, and rely on the left sidebar for navigation and less duplication of labels. Cc @AlexTugarev 
 
| Example of labels repeating |
|-|
| Frame 1303 |

File: components/dashboard/src/teams/GitIntegrationsPage.tsx
Comment (gtsiolis) [inline components/dashboard/src/teams/GitIntegrationsPage.tsx:13]: **thought**: We can probably update the copy in the empty state for Git integrations, but we can make these in a separate PR.

Comment (gtsiolis) [inline components/dashboard/src/teams/GitIntegrationsPage.tsx:13]: **praise**: This is out of the scope of the changes in this PR, but wanted to say thanks for improving the copy in https://github.com/gitpod-io/gitpod/pull/16850 from the previous iteration. 🌮 🌮

File: components/dashboard/src/components/Modal.tsx
Comment (gtsiolis) [reply components/dashboard/src/components/Modal.tsx:153]: **issue**: Using the alert component here sounds perfect, however, using alerts or error feedback inside modals can be tricky, especially after adding a scrollable area inside the modal in https://github.com/gitpod-io/gitpod/pulls?page=3&q=is%3Apr+author%3Aselfcontained+is%3Aclosed.

Comment (gtsiolis) [inline components/dashboard/src/components/Modal.tsx:153]: **suggestion**: Since we're having validation inline for text input, we could do the following in this or a follow PR:
1. Skip the alert here and rely on the SSO clients list to communicate an error or a failed attempt.
2. Migrate away from the modals to a full page layout as we do with the access tokens flow, which is more appropriate, scalable, and accessible.

| Input validation | Error feedback through list | Error feedback in full page |
|-|-|-|
| Screenshot 2023-03-16 at 14 36 12 | Screenshot 2023-03-16 at 14 36 46 | Screenshot 2023-03-16 at 14 36 58 |

File: components/dashboard/src/teams/sso/OIDCClients.tsx
Comment (gtsiolis) [inline components/dashboard/src/teams/sso/OIDCClients.tsx:59]: **thought**: I'm sure we can come up with a better and more readable copy here, but let's do that in a separate PR.

**Review by gtsiolis — APPROVED** [review]
UX LGTM! 

Additionally approving to unblock merging and holding in case we'd like to address any of the UX issues in this PR. 🏓 

/hold

File: components/dashboard/src/teams/sso/OIDCClientConfigModal.tsx
Comment (selfcontained) [reply components/dashboard/src/teams/sso/OIDCClientConfigModal.tsx:92]: Yah, I don't have the modal-based error message component built in a way that supports a dynamic amount of content very well atm, but I agree, we should be able to include those details in the error message, just need to make it more robust so it can expand to fit content, but also not block the modal body content it slides over. I'll tackle that in a follow-up.

File: components/dashboard/src/components/Modal.tsx
Comment (selfcontained) [reply components/dashboard/src/components/Modal.tsx:153]: Yup, I think having the alert positioned absolutely somewhere so it's kind of pinned at the footer is one way to avoid some of those issues (but does introduce some other challenges). Currently I'm using a pattern where the alert is pinned right above the footer (part of the footer actually). I've added some extra padding to the bottom of the modal body section to account for it, but that's a temporary solution and not ideal. What I'd like to do:

* Animate the alert so it slides up from the footer.
* Find a more robust way to deal with the alert potentially blocking content in the modal body. Some ideas:
  * Dynamically adding padding to the bottom of the modal body based on the size of the alert when it's present so content can be scrolled to in the body.
  * Making the pinned modal alert dismissible (probably the easiest solution to implement).

File: components/dashboard/src/teams/GitIntegrationsPage.tsx
Comment (selfcontained) [reply components/dashboard/src/teams/GitIntegrationsPage.tsx:12]: Oh, that's my mistake then, I understood some of that differently. I can change this back if that's what we want in a followup pretty easily. It is kinda awkward coming up with 3 versions of the page title that are slightly different, so this would remove that 😄

File: components/dashboard/src/components/Modal.tsx
Comment (selfcontained) [reply components/dashboard/src/components/Modal.tsx:153]: Good points. 

If we stuck with the modal, we'd need some version of an alert for any error messages returned from the api call during create/update.

I'd be on board with using full pages here. I debated going down that route, but felt like a bigger change I didn't want to introduce quite yet. It would also simplify some of the complexity in the git auth modal that deals with managing state after creation.

Either way, I think having a way to show simple error messages for a modal in a consistent way will helpful, but need to address some of the usability points we've discussed.

Comment (gtsiolis) [reply components/dashboard/src/components/Modal.tsx:153]: over-here

**Pull request merged**
